samigo/samigo-services/src/java/org/sakaiproject/tool/assessment/services/assessment/AssessmentEntityProducer.java

```diff
@@ -393,4 +520,284 @@ public void updateEntityReferences(String toContext, Map<String, String> transve
+		String[] attachmentParts = attachmentRef.split("\\|");
+		String realName = attachmentParts[1];
+		String attachmentId = attachmentParts[0].replace("/Tests_Quizzes/","/Tests _ Quizzes/").replace(realName.replace(" ",""), realName);
+
+		result.add(attachmentId);
+		log.info("Found attachment: {}", attachmentId);
+	}
+
+	return result;
+    }
+
+    /*
+     * Parse an HTML text blob to extract inline URLs and turn them back into
+     * Sakai references. Similar to
+     * https://github.com/cilt-uct/sakai/blob/21.x/lessonbuilder/tool/src/java/org/sakaiproject/lessonbuildertool/service/LessonBuilderEntityProducer.java#L470
+     */
+    private List<String> parseInlineResourceRefs(String siteId, String qText) {
+
+	if (StringUtils.isEmpty(qText) || !qText.contains("<") || qText.equals("<![CDATA[]]>")) {
+		return Collections.emptyList();
+	}
+
+	if (qText.contains("CDATA")) {
+		qText = qText.replace("<![CDATA[", "").replace("]]>", "");
+	}
+
+        List<String> result = new ArrayList<>();
+	org.jsoup.nodes.Document htmlDoc = Jsoup.parse(qText);
+
+        // Typically audio or video <source> or <img>
+	Elements media = htmlDoc.select("[src]");
+
+	for (org.jsoup.nodes.Element src : media) {
+		String link = src.attr("abs:src");
+
+		// embedded fckeditor attachments
+		if (link.contains("/access/content/attachment/")) {
+			String linkRef = link.replace(link.substring(0, link.indexOf("/attachment/")), "");
+			result.add(linkRef);
+			log.info("Found inline attachment asset: {} adding to attachment list as: {}", link, linkRef);
+			continue;
+		}
+
+		// URLs below may include spaces, so URLDecode to turn back into a Sakai content ID
+		try {
+			// cross-site references
+			if (link.contains("/access/content/group/") && !link.contains(siteId)) {
+				String linkRef = URLDecoder.decode(link.replace(link.substring(0, link.indexOf("/group/")), ""), "UTF-8");
+				result.add(linkRef);
+				log.info("Found inline cross-site asset: {} adding to attachment list as: {}", link, linkRef);
+			}
+
+			// references to user workspace files: map the user eid in the path to id
+			if (link.contains("/access/content/user/")) {
+				String linkRef = URLDecoder.decode(link.replace(link.substring(0, link.indexOf("/user/")), ""), "UTF-8");
+				String linkParts[] = linkRef.split("/", 4);
+				String userId = UserDirectoryService.getUserId(linkParts[2]);
+				linkRef = "/user/" + userId + "/" + linkParts[3];
+				result.add(linkRef);
+				log.info("Found inline user workspace asset: {} adding to attachment list as: {}", link, linkRef);
+			}
+
+			// references to public files
+			if (link.contains("/access/content/public/")) {
+				String linkRef = URLDecoder.decode(link.replace(link.substring(0, link.indexOf("/public/")), ""), "UTF-8");
+				result.add(linkRef);
+				log.info("Found inline public asset: {} adding to attachment list as: {}", link, linkRef);
+			}
+
+		} catch (UnsupportedEncodingException e) {
```

Comment:
Only one catch with both and the same message?

log.warn("Unable to add link {} to attachment list: {}", link, e.getMessage());